build/ChoosePowerShell.cmd

```diff
@@ -0,0 +1,23 @@
+:: where.exe does not exist in windows container, application specific test must be used to check for existence
+
+PowerShell -Command Write-Host "a"
```

Comment:
Is there a scenario where we would use the `PowerShell` instead of `pwsh`?

Reply:
Building on dev machine. Our dev machines typically have powershell by default but not pwsh. Wouldn't want to block someone from cloning and building.

Reply:
Okay, makes sense. Since `pwsh` is a newer version, should we attempt to use it first, followed by `PowerShell` in case the former isn't available?

Reply:
We could, but I'm not sure why we would. If the order was important then the design to use whichever one is available would be flawed.

Reply:
My understanding is if the user has both `PowerShell 4.0` and `pwsh 6.0` installed, then this script would use the former, even though the latter is a newer version. In this case, the order does influence the version used for execution.

Reply:
More people will have 3.1 SDK installed on their dev box over time, so people are more likely to use the same "tool" on dev and build machines. It's good to be consistent. Maybe it's a good idea to print out what is picked for easier debugging in case they do present a behavior difference one day.

Reply:
That's right. But I'm trying to see if there is a concern with using PowerShell 4.0 even if pwsh 6.0 is installed. If there was, then the approach that I went with that either one could be used for building is invalid.

Reply:
I'll reverse the order. We use pwsh in the official build. If the official build fails for any reason and we suspect pwsh as the cause then it will be easier to debug on dev machine if pwsh is first.